Report/sections/2_design/semantics.tex

```diff
@@ -66,6 +66,14 @@ \subsection*{Type rules in \dazel{}}
 \ref{item:tiletype} is an important type: it is the type behind every tile in the generated games.
 \ref{item:arraytype} means that every type in the list will have a corresponding array type. This is to say that arrays are parametric - for example, an array could be an array of numbers.
 
+\subsection*{Dynamic and static binding}
+A variable must first be bound to a data type before it can be used in the program. 
+This is called a binding and, depending on the language, it can either be dynamic or static.
+\textbf{Dynamic binding} is when the type of a variable is determined at runtime. 
+\textbf{Static binding} is when the type of a variable can be determined during compilation as all the necessary information is present in the source code \cite{sebesta_concepts_2016}.
+
+\dazel{} uses static binding as types are determined while the source code is compiled to API calls. When a variable is initially declared, it is assigned a type before the code is run.
```

Comment:
We shouldn't write 'compiled' when we're building an interpreter. Typechecking can be mentioned here - that's where we determine which types variables have.

Reply:
The code still gets compiled into some intermediate code though. The intermediate code is then what is interpreted.